### ISSUE TITLE
Integrate the watchtower-plugin into CLN on Umbrel

### ISSUE THREAD

**Issue by mariocynicys** [talaia-labs/rust-teos#207]
Umbrel is what a lot of lightning node runners host their nodes on, because it has an easy to use interface and it's easy to install applications on.
Applications on Umbrel are like applications on an app store, and these applications can be any app you can think of, running inside a docker container, and has a web interface that is accessible from one's Umbrel endpoint.

There is a Core-Lightning (CLN) application available on Umbrel to run CLN nodes.
We can add the `watchtower-plugin` to Umbrel and integrate it with CLN.

I think this could be done in one of two ways:
- Make a stand-alone app on Umbrel that would somehow add itself to the already installed CLN app and make it run the `watchtower-plugin`.
- Make a simple button and maybe a stats window to the already published CLN app that would do the same.

**Comment by sr-gi** [comment]
I think this is a great idea. I'd link it to the SoB project about building the GUI for `teos`, they may get along well.

**Comment by sr-gi** [comment]
Related to #192

**Comment by anipaul2** [comment]
Hi @sr-gi ,
Are there any specific design guidelines or requirements for creating a new app or modifying an existing one on Umbrel that I should be aware of before proceeding?

**Comment by sr-gi** [comment]
Nope, we haven't worked with Umbrel before, so all needs to be done from scratch.

Using an existing app as the base is an option, and most likely recommend.

**Comment by anipaul2** [comment]
> Nope, we haven't worked with Umbrel before, so all needs to be done from scratch.
> 
> Using an existing app as the base is an option, and most likely recommend.

okay.

**Comment by jonmarrs** [comment]
Why not make a watchtower-plugin Umbrel app that works with LND as well?

**Comment by sr-gi** [comment]
For that to work we'd need a way to pull the required information from LND, however, LND does not expose that publicly, so currently there is no way of doing so.

**Comment by jonmarrs** [comment]
So you're basically saying that watchtowers only work for CLN, not LND?

**Comment by mariocynicys** [comment]
> So you're basically saying that watchtowers only work for CLN, not LND?

I beleive LND has their built-in 3rd party watchtower. The watchtower plugin though only works for CLN.

**Comment by jonmarrs** [comment]
Interesting. I wonder if we could request that LND make the required interface public.

**Comment by sr-gi** [comment]
> So you're basically saying that watchtowers only work for CLN, not LND?

No, that's certainly not what I'm saying. LND does has it's own watchtower, but it uses a private interface to interact with the node that third party software has no access to.

> Interesting. I wonder if we could request that LND make the required interface public.

I've tried to talk them into it for years, but I encourage you to try. Maybe the more people requesting it the better.

**Comment by moneyball** [comment]
Is there an open LND GitHub issue detailing what is needed to make TEOS work with LND?

**Comment by sr-gi** [comment]
> Is there an open LND GitHub issue detailing what is needed to make TEOS work with LND?

I don't think there is, but I'm happy to open one at this point.

**Comment by sr-gi** [comment]
https://github.com/lightningnetwork/lnd/issues/7680 @moneyball
